tests/test_categorical_encoding.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+import pandas as pd
+from src.koheesio.pandas.categorical_encoding  import PandasCategoricalEncoding
+import unittest
+import pandas as pd
+
+class TestPandasCategoricalEncoding(unittest.TestCase):
```

Comment:
don't use unittest - use pytest.
1. get rid of the unittest.TestCase (just let it be a regular class)
2. change your `self.assert...` (from unittest) to regular python `assert` (this is how pytest works)
3. get rid of your setUp - just make the input dataframe a module level variable, OR make it a fixture (a bit overkill for your purpose here)
4. change your code to match the interface I proposed above